packages/flutter/lib/src/material/elevation_overlay.dart

```diff
@@ -49,7 +49,7 @@ class ElevationOverlay {
     if (elevation > 0.0 &&
         theme.applyElevationOverlayColor &&
         theme.brightness == Brightness.dark &&
-        color == theme.colorScheme.surface) {
+        color.withOpacity(1.0) == theme.colorScheme.surface.withOpacity(1.0)) {
```

Comment:
This test is getting more and more complicated. Seems like it is just a band-aid over the fact that we really don't have a concept of what is a surface and what isn't. That said, this is a good fix for now. Thx.

Reply:
Agreed, we are kind of patching things up. Material-Android has gotten to a stable place with dark theme logic, so hopefully once we've patched the same holes that they have, we will be able to leave it alone. Removing opacities is another part of their check.